File: Withdraw Money/withdrawMoney.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int n, S;
vector<int> t;
int Count = 0;
vector<vector<int>> ways;

void withdrawMoney(int id, int total, vector<int> path)
{
    if (total == S)
    {
        ways.push_back(path);
        Count++;
        return;
    }

    for (int i = id; i < n; i++)
    {
        if (t[i] + total <= S)
        {
            path.push_back(t[i]);
            withdrawMoney(i, total + t[i], path);
            path.pop_back();
        }
    }
}

int main()
{
    cin >> n >> S;
    t.resize(n);
    for (int i = 0; i < n; i++)
    {
        cin >> t[i];
    }

    sort(t.begin(), t.end());

    withdrawMoney(0, 0, {});

    cout << Count << endl;
    for (auto way : ways)
    {
        for (auto num : way)
        {
            cout << num << " ";
        }
        cout << endl;
    }

    return 0;
}

File: Withdraw Money/WithdrawMoney.py
def withdrawMoney(id, total, path):
    global Count
    if total == S:
        ways.append(path)
        Count = Count + 1
        return True

    for i in range(id, n):
        if t[i] + total <= S:
            withdrawMoney(i, total + t[i], path + [t[i]])


# Read input from a file
with open('input10.txt', 'r') as f:
    n, S = map(int, f.readline().split())
    t = list(map(int, f.readline().split()))

t.sort()
Count = 0
ways = []
withdrawMoney(0, 0, [])

# Write output to a file
with open('output10.txt', 'w') as f:
    f.write(str(Count) + '\n')
    for way in ways:
        f.write(' '.join(map(str, way)) + '\n')
